homeassistant/components/shelly/utils.py

```diff
@@ -220,12 +222,27 @@ def get_shbtn_input_triggers() -> list[tuple[str, str]]:
 async def get_coap_context(hass: HomeAssistant) -> COAP:
     """Get CoAP context to be used in all Shelly Gen1 devices."""
     context = COAP()
+
+    adapters = await network.async_get_adapters(hass)
+    LOGGER.debug("Network adapters: %s", adapters)
+
+    ipv4: list[IPv4Address] = []
+    if not network.async_only_default_interface_enabled(adapters):
+        for address in await network.async_get_enabled_source_ips(hass):
+            if address.version == 4 and not (
+                address.is_link_local
+                or address.is_loopback
+                or address.is_multicast
+                or address.is_unspecified
+            ):
+                ipv4.append(address)
+    LOGGER.debug("Network IPv4 addresses: %s", ipv4)
     if DOMAIN in hass.data:
         port = hass.data[DOMAIN].get(CONF_COAP_PORT, DEFAULT_COAP_PORT)
     else:
         port = DEFAULT_COAP_PORT
     LOGGER.info("Starting CoAP context with UDP port %s", port)
-    await context.initialize(port)
+    await context.initialize(port, ipv4)
```

Comment:
I think we should have a test that wraps `context.initialize` checks what is passed to it

Reply:
I added a test with 2 ip addresses, while the others use the default one.
We can always expand it, but I need some help ;-)

Reply:
I think you want to patch it with `wraps` https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock and than check what it sends

> wraps: Item for the mock object to wrap. If wraps is not None then calling the Mock will pass the call through to the wrapped object (returning the real result). Attribute access on the mock will return a Mock object that wraps the corresponding attribute of the wrapped object (so attempting to access an attribute that doesn’t exist will raise an [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError)).